### PR TITLE
1016 - Implementing the redesign of the instructor teaching call form

### PR DESCRIPTION
**Pull request by ltwheeler**

### CODE REVIEW

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 1]
```diff
@@ -92,8 +92,56 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
 			};
 
-			$scope.updateAssignmentsOrder = function(sortedTeachingPreferenceIds, termContainer) {
-				teachingCallFormActionCreators.updateAssignmentsOrder(sortedTeachingPreferenceIds, $scope.view.state.scheduleId, termContainer.termCode);
+			$scope.raisePriority = function (preference, preferences, termCode) {
+				var sortedTeachingPreferenceIds = [];
+				var indexToSwap = null;
+
+				// Sort by priority
+				var sortedPreferences = _array_sortByProperty(preferences, "priority");
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 2]
```diff
@@ -92,8 +92,56 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
+
+				// Construct preferenceId array
+				for (var i = 0; i < sortedPreferences.length; i++) {
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: Array.forEach() or Array.map() candidate?

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js [hunk 3]
```diff
@@ -92,8 +92,56 @@ teachingCallApp.controller('TeachingCallFormCtrl', ['$scope', '$rootScope', '$wi
...
+					var slotPreference = sortedPreferences[i];
+					sortedTeachingPreferenceIds.push(slotPreference.id);
+
+					if (slotPreference.id == preference.id) {
+						indexToSwap = i;
+					}
+				}
+
+				var temp = sortedTeachingPreferenceIds[indexToSwap-1];
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: indexToSwap(space)-(space)1 is more readable

File: app/teachingCall/teachingCallForm/services/teachingCallFormStateService.js [hunk 4]
```diff
@@ -85,6 +85,28 @@ teachingCallApp.service('teachingCallFormStateService', function (
 						termData.preferenceOptions = self.generatePreferenceOptions(action.payload.scheduleId, action.payload.instructorId, termData.termCode, termData.preferences, termData.assignments, pageState.sectionGroups, pageState.coursesIndex);
 					});
 
+					// Set selectedTermCode
+					pageState.selectedTermCode = pageState.terms[0].termCode;
+
+					// Calculate termSelection UI
+					pageState.termSelection = [];
+
+					pageState.terms.forEach ( function (term) {
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/services/teachingCallFormStateService.js]: pageState.terms.forEach( (term) => {

File: app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html [hunk 5]
```diff
@@ -0,0 +1,184 @@
+<nav year="{{ year }}" workgroup-id="{{ workgroupId }}"></nav>
+
+<ipa-header page-title="Teaching Call"></ipa-header>
+
+<div ng-hide="view.state" style="margin-top:300px;">
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html]: Should the 'style' be moved to a CSS file?

File: app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+	<spinner></spinner>
+</div>
+
+
+<!-- handle lack of active teaching call -->
+<div ng-if="view.state && !view.state.isInstructorInTeachingCall">
+		<div class="jumbotron" style="text-align: center; height: 100%; background-color: white;">
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html]: CSS file usage?

File: app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+						{{ term.termDescription }}
+					</div>
+					<span style="padding-left: 10px;" class="glyphicon glyphicon-transfer text-primary clickable hovrable" tooltip-placement="bottom" uib-tooltip="Copy to all terms"
+						confirm-button="copyUnavailabilitiesToAllTerms(term.availabilityBlob)"
+						message="Are you sure you want to copy this term availabilities to all other terms?
+							This will overwrite existing availabilities."
+						yes="Confirm" no="Cancel" placement="top"></span>
+				</div>
+				<availability-grid blob="term.availabilityBlob"
+					on-change="saveTeachingCallResponse(term, blob, 3000)">
```
Comment (cthielen) [inline app/teachingCall/teachingCallForm/templates/TeachingCallForm-old.html]: 3000 feels like a magic number to me.

File: app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js (hunk 1 above)
Comment (ltwheeler) [inline app/teachingCall/teachingCallForm/controllers/TeachingCallFormCtrl.js]: _array_sortByProperty makes use of the Array.sort method, it just has additional checks/ common quality of life checks.

File: .eslintrc [hunk 8]
```diff
@@ -1,10 +1,17 @@
...
 		"node": true,
 		//enable browser environment
 		"browser": true
 	},
+  "parserOptions": {
+    "ecmaVersion": 6,
+  },
+  "ecmaFeatures": {
+    "arrowFunctions":true
+  },
```
Comment (cthielen) [inline .eslintrc]: Please fix the spacing to be consistent.

**Pull request merged**